java/source/flight.rst

```diff
@@ -0,0 +1,1699 @@
+        private final BufferAllocator allocator;
+        private Location location;
+
+        public InMemoryStore(BufferAllocator allocator, Location location) {
+            super();
+            this.allocator = allocator;
+            this.location = location;
+        }
+
+        public void setLocation(Location location) {
+            this.location = location;
+        }
+
+        @Override
+        public void getStream(CallContext context, Ticket ticket,
+                              FlightProducer.ServerStreamListener listener) {
+            System.out.println("Calling to getStream");
+            getStream(ticket).sendTo(allocator, listener);
+        }
+
+        public Stream getStream(Ticket t) {
+            ExampleTicket example = ExampleTicket.from(t);
+            FlightDescriptor d = FlightDescriptor.path(example.getPath());
+            FlightHolder h = holders.get(d);
+            if (h == null) {
+                throw new IllegalStateException("Unknown ticket.");
+            }
+
+            return h.getStream(example);
+        }
+
+        @Override
+        public void listFlights(CallContext context, Criteria criteria, StreamListener<FlightInfo> listener) {
+            System.out.println("Calling to listFligths");
+            try {
+                for (FlightHolder h : holders.values()) {
+                    listener.onNext(h.getFlightInfo(location));
+                }
+                listener.onCompleted();
+            } catch (Exception ex) {
+                listener.onError(ex);
+            }
+        }
+
+        @Override
+        public FlightInfo getFlightInfo(CallContext context, FlightDescriptor descriptor) {
+            System.out.println("Calling to getFlightInfo");
+            FlightHolder h = holders.get(descriptor);
+            if (h == null) {
+                throw new IllegalStateException("Unknown descriptor.");
+            }
+
+            return h.getFlightInfo(location);
+        }
+
+        @Override
+        public Runnable acceptPut(CallContext context,
+                                  final FlightStream flightStream, final StreamListener<PutResult> ackStream) {
+            return () -> {
+                System.out.println("Calling to acceptPut");
+                StreamCreator creator = null;
+                boolean success = false;
+                try (VectorSchemaRoot root = flightStream.getRoot()) {
+                    final FlightHolder h = holders.computeIfAbsent(
+                            flightStream.getDescriptor(),
+                            t -> new FlightHolder(allocator, t, flightStream.getSchema(), flightStream.getDictionaryProvider()));
+
+                    creator = h.addStream(flightStream.getSchema());
+
+                    VectorUnloader unloader = new VectorUnloader(root);
+                    while (flightStream.next()) {
+                        ackStream.onNext(PutResult.metadata(flightStream.getLatestMetadata()));
```

Comment:
IMO we can just ignore metadata for the sake of this example

Reply:
Deleted